### ISSUE TITLE
Unhandled exception in missing_non_null_constraint

### ISSUE THREAD

**Issue by ariccio** [gregnavis/active_record_doctor#53]
It looks like ARD crashes when some relations don't exist:

```
C:\Users\Lucius Riccio\Documents\GitHub\COVID-CO2-tracker\COVID_CO2_TRACKER>bundle exec rake active_record_doctor:missing_non_null_constraint --trace
** Invoke active_record_doctor:missing_non_null_constraint (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute active_record_doctor:missing_non_null_constraint
rake aborted!
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "action_text_rich_texts" does not exist
LINE 8:  WHERE a.attrelid = '"action_text_rich_texts"'::regclass
                            ^
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `exec'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `block (2 levels) in query'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:18:in `block in query'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract_adapter.rb:696:in `block (2 levels) in log'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract_adapter.rb:695:in `block in log'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract_adapter.rb:687:in `log'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:17:in `query'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql_adapter.rb:824:in `column_definitions'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:116:in `columns'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/missing_non_null_constraint.rb:14:in `block in run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/missing_non_null_constraint.rb:11:in `map'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/missing_non_null_constraint.rb:11:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/base.rb:7:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/tasks/active_record_doctor.rake:17:in `block in mount'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `block in execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:188:in `invoke'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:160:in `invoke_task'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `block (2 levels) in top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `block in top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:125:in `run_with_threads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:110:in `top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:83:in `block in run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:186:in `standard_exception_handling'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:80:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
C:/Ruby30-x64/bin/rake:31:in `load'
C:/Ruby30-x64/bin/rake:31:in `<main>'

Caused by:
PG::UndefinedTable: ERROR:  relation "action_text_rich_texts" does not exist
LINE 8:  WHERE a.attrelid = '"action_text_rich_texts"'::regclass
                            ^
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `exec'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `block (2 levels) in query'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:18:in `block in query'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract_adapter.rb:696:in `block (2 levels) in log'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract_adapter.rb:695:in `block in log'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activesupport-6.1.2.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract_adapter.rb:687:in `log'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:17:in `query'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/postgresql_adapter.rb:824:in `column_definitions'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/activerecord-6.1.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:116:in `columns'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/missing_non_null_constraint.rb:14:in `block in run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/missing_non_null_constraint.rb:11:in `map'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/missing_non_null_constraint.rb:11:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/active_record_doctor/tasks/base.rb:7:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/active_record_doctor-1.7.1/lib/tasks/active_record_doctor.rake:17:in `block in mount'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `block in execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in `execute'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `synchronize'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in `invoke_with_call_chain'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/task.rb:188:in `invoke'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:160:in `invoke_task'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `block (2 levels) in top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `each'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in `block in top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:125:in `run_with_threads'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:110:in `top_level'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:83:in `block in run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:186:in `standard_exception_handling'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/lib/rake/application.rb:80:in `run'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
C:/Ruby30-x64/bin/rake:31:in `load'
C:/Ruby30-x64/bin/rake:31:in `<main>'
Tasks: TOP => active_record_doctor:missing_non_null_constraint
```

**Comment by gregnavis** [comment]
Thank you for reporting. This is fixed in 14aba5697dfcc5e7647817376bfe92dbff0c40d8. The fix should be available later this week after I release 1.7.2.

**Comment by gregnavis** [comment]
I've just released 1.7.2 with the aforementioned fix. I'm closing the issue but if there are still issues feel free to reopen.

**Comment by ariccio** [comment]
Thank you. I'm quite glad someone wrote something to statically check these issues! It's an excellent little tool, something I don't see enough of in web dev.
